test/jit/test_list_dict.py

```diff
@@ -892,6 +892,141 @@ def check_list(fn, li):
+
+        def to_list_int_1D(x):
+            # type: (torch.Tensor) -> List[int]
+            li : List[int] = x.tolist()
+            return li
+
+        def to_list_int_2D(x):
+            # type: (torch.Tensor) -> List[List[int]]
+            li : List[List[int]] = x.tolist()
+            return li
+
+        def to_list_int_3D(x):
+            # type: (torch.Tensor) -> List[List[List[int]]]
+            li : List[List[List[int]]] = x.tolist()
+            return li
+
+        self.checkScript(to_list_int_0D, (torch.tensor(1, dtype=torch.long),))
+        int_input_1D = torch.tensor([1, 2, 3, 4], dtype=torch.long)
+        self.checkScript(to_list_int_1D, (int_input_1D,))
+        int_input_2D = torch.tensor([[1, 2, 3], [3, 4, 5]], dtype=torch.long)
+        self.checkScript(to_list_int_2D, (int_input_2D,))
+        int_input_3D = torch.tensor([[[1, 2], [3, 4]], [[5, 6], [7, 8]]], dtype=torch.long)
+        self.checkScript(to_list_int_3D, (int_input_3D,))
+        int_input_noncontiguous = torch.tensor([[[1, 2], [3, 4]], [[5, 6], [7, 8]]], dtype=torch.long).transpose(0, 1)
+        self.checkScript(to_list_int_3D, (int_input_noncontiguous,))
+
+        """
+        Float dtype unit tests.
+        """
+        def to_list_float_0D(x):
+            # type: (torch.Tensor) -> float
+            li : float = x.tolist()
+            return li
+
+        def to_list_float_1D(x):
+            # type: (torch.Tensor) -> List[float]
+            li : List[float] = x.tolist()
+            return li
+
+        def to_list_float_2D(x):
+            # type: (torch.Tensor) -> List[List[float]]
+            li : List[List[float]] = x.tolist()
+            return li
+
+        def to_list_float_3D(x):
+            # type: (torch.Tensor) -> List[List[List[float]]]
+            li : List[List[List[float]]] = x.tolist()
+            return li
+
+        self.checkScript(to_list_float_0D, (torch.randn(5)[0],))
+        self.checkScript(to_list_float_1D, (torch.randn(5),))
+        self.checkScript(to_list_float_2D, (torch.randn(5, 6),))
+        self.checkScript(to_list_float_3D, (torch.randn(5, 6, 7),))
+        self.checkScript(to_list_float_3D, (torch.randn(5, 6, 7).transpose(0, 1),))
+
+        """
+        Non-happy path tests:
+            - missing type annotation
+            - mismatch between type annotation and input
+            - type annotation with unsupported type
+            - input with unsupported type
+        """
+        def to_list_missing_type_annotation(x):
+            # type: (torch.Tensor) -> List[float]
+            li = x.tolist()
+            return li
+
+        def to_list_incorrect_type_annotation(x):
+            # type: (torch.Tensor) -> List[float]
+            li : float = x.tolist()
+            return li
+
+        def to_list_unsupported_type_annotation(x):
+            # type: (torch.Tensor) -> List[float]
+            li : List[str] = x.tolist()
+            return li
+
+        def to_list_unsupported_input_type(x):
```

Comment:
i think you forgot to compile this one

Reply:
Meant to delete this because I cannot create a `Tensor` of a type that `tolist` does not support